src/model.js

```diff
@@ -0,0 +1,278 @@
+      };
+      callback(null, featureCollection);
+    },
+    (err) => {
+      callback(err);
+    }
+  );
+};
+
+function search(query) {
+  return new Promise((resolve, reject) => {
+    // If this service is added to an ArcGIS Online Map, usually it makes multiple
+    // calls (4 quadrants). When you make 4 calls to the Yelp API at the same
+    // time, you get a Yelp error, so for now set a random pause 0.5-2 seconds.
+    // TODO: Remove this random pause code, and change it so that if we get a
+    // "TOO_MANY_REQUESTS_PER_SECOND" error, retry after a random pause.
+
+    const randomPauseTime = Math.random() * (2500 - 500) + 500;
+
+    setTimeout(() => {
+      searchYelp(query).then(
+        (features) => {
+          resolve(features);
+        },
+        (err) => {
+          reject(err);
+        }
+      );
+    }, randomPauseTime);
+  });
+}
+
+// Wrap the call to Yelp, this will make testing easier and decouple us from the specific client lib
+function searchYelp(query) {
+  return new Promise((resolve, reject) => {
+    client.search(query).then(
+      (rawResponse) => {
+        const features = translate(rawResponse.jsonBody, query.term);
+        resolve(features);
+      },
+      (err) => {
+        reject(err);
+      }
+    );
+  });
+}
+
+// Translate a request from the GeoServices API into something Yelp will understand
+function buildQuery(options) {
+  // we don't want to modify the passed in options object because Koop will use that later
+  // also for some reason the Yelp lib chokes when there is a passed in callback so omit it
+  const query = _.omit(_.cloneDeep(options), "callback");
+  if (
+    options.geometryType &&
+    options.geometry &&
+    options.geometryType === "esriGeometryEnvelope"
+  ) {
+    const bbox = options.geometry;
+    [query.longitude, query.latitude] = getCenter(bbox);
+    query.radius = getRadius(bbox);
+  } else if (!options.location) {
+    query.location = "St. Louis, MO";
+  }
+  query.term = setTerm(options);
+  const sort = setSort(options);
+  if (sort) {
+    query.sort_by = sort;
+  }
+
+  query.limit = 50;
+  return query;
+}
+
+/**
+ * Given a bounding box (assuming web mercator right now), finds the center
+ * and returns [longitude, latitude]
+ *
+ * @param {object} bbox
+ */
+function getCenter(bbox) {
+  const [xmin, ymin, xmax, ymax] = getBBox(bbox);
+
+  const x = (xmax + xmin) / 2;
+  const y = (ymax + ymin) / 2;
+
+  return proj.forward([x, y]);
+}
+
+/**
+ * Input bbox may be in full JSON or shortened version. This function
+ * normalizes to give same results for both formats. Returns 0,0 if bad format.
+ * @param {string | object} bbox
+ */
+function getBBox(bbox) {
+  let xmin, ymin, xmax, ymax;
+
+  if (typeof bbox === "string") {
+    const parts = bbox.split(",");
+    if (parts.length === 4) {
+      [xmin, ymin, xmax, ymax] = parts.map((x) => parseFloat(x));
+    } else {
+      // invalid format, set to 0,0
+      console.error("Invalid format");
```

Comment:
I don't want to hold this up any longer, but if the geometry parameter is invalid, it should return a 400 error to the client.  Just add a TODO for now.

Reply:
fixed